Scripts/UtilityScripts/AzTSUtility.ps1

```diff
@@ -0,0 +1,1617 @@
+    The FTP State to set. Either "Disabled" or "FtpsOnly". Defaults to "Disabled".
+    .INPUTS
+    None
+    .OUTPUTS
+    None
+    .EXAMPLE
+    PS> Set-AppServiceFtpState -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -AppServiceName "MyAppServiceName" -SlotName "MyNonProductionSlotName" -FtpState "Disabled"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ResourceGroupName,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $AppServiceName,
+    [Parameter(Mandatory = $false)]
+    [string]
+    $SlotName,
+    [Parameter(Mandatory = $false)]
+    [string]
+    $FtpState = "Disabled"
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  if ([string]::IsNullOrWhiteSpace($SlotName))
+  {
+    # No slot name so we're setting the app service itself
+    Set-AzWebApp -ResourceGroupName $ResourceGroupName -Name $AppServiceName -FtpsState $FtpState
+  }
+  else
+  {
+    if ($SlotName.StartsWith($AppServiceName))
+    {
+      $SlotName = $SlotName.Replace($AppServiceName + "/", "")
+    }
+
+    Set-AzWebAppSlot -ResourceGroupName $ResourceGroupName -Name $AppServiceName -Slot $SlotName -FtpsState $FtpState
+  }
+}
+
+# ####################################################################################################
+
+# ####################################################################################################
+# Azure_DataFactory_DP_Avoid_Plaintext_Secrets
+
+function Get-DataFactoryV2()
+{
+  <#
+    .SYNOPSIS
+    This command shows the data factory.
+    .DESCRIPTION
+    This command shows the data factory.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the Data Factory.
+    .PARAMETER ResourceGroupName
+    The Resource Group name containing the Data Factory.
+    .PARAMETER DataFactoryName
+    The Data Factory name.
+    .INPUTS
+    None
+    .OUTPUTS
+    Text with the Data Factory name and its Tag names/values.
+    .EXAMPLE
+    PS> Get-DataFactoryV2 -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -DataFactoryName "MyDataFactoryName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+      [Parameter(Mandatory=$true)]
+      [string]
+      $SubscriptionId,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $ResourceGroupName,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $DataFactoryName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $factory = Get-AzDataFactoryV2 -ResourceGroupName $ResourceGroupName -Name $DataFactoryName
+
+  Write-Information -InformationAction Continue -MessageData ("Data Factory: " + $factory.DataFactoryName)
+  Write-Information -InformationAction Continue -MessageData "Tags:"
+  Write-Information -InformationAction Continue -MessageData $factory.Tags
```

Comment:
Please validate, this line does not printed tags in console for me, instead printed just this: System.Collections.Generic.Dictionary`2[System.String,System.String]

Reply:
Mitigated - implemented logic to iterate through Tags hashtable.